structure_tests/structure_test.go

```diff
@@ -88,39 +87,48 @@ func Parse(t *testing.T, fp string) (StructureTest, error) {
 	if !ok {
 		return nil, errors.New("Error encountered when type casting Structure Test interface")
 	}
-	tests.SetDriverImpl(driverImpl, imageName)
+	tests.SetDriverImpl(driverImpl, imagePath)
 	return tests, nil
 }
 
 var configFiles arrayFlags
-var imageName, driver string
-var driverImpl func(string) drivers.Driver
+
+var imagePath, driver string
+var driverImpl func(string) (drivers.Driver, error)
 
 func TestMain(m *testing.M) {
-	flag.StringVar(&imageName, "image", "", "path to test image")
+	flag.StringVar(&imagePath, "image", "", "path to test image")
 	flag.StringVar(&driver, "driver", "docker", "driver to use when running tests")
 
+	var logLevel string
+	flag.StringVar(&logLevel, "logLevel", "0", "test")
+	flag.Lookup("v").Value.Set(logLevel)
+
 	flag.Parse()
 	configFiles = flag.Args()
-	stdout := bufio.NewWriter(os.Stdout)
 
-	if imageName == "" {
-		fmt.Fprintln(stdout, "Please supply name of image to test against")
+	if imagePath == "" {
+		fmt.Println("Please supply path to image or tarball to test against")
 		os.Exit(1)
 	}
 
 	if len(configFiles) == 0 {
-		fmt.Fprintln(stdout, "Please provide at least one test config file")
+		fmt.Println("Please provide at least one test config file")
 		os.Exit(1)
 	}
 
 	var err error
-	driverImpl, err = drivers.InitDriverImpl(driver)
+
+	driverImpl = drivers.InitDriverImpl(driver)
+	if driverImpl == nil {
+		fmt.Printf("Unsupported driver type: %s", driver)
+		os.Exit(1)
+	}
 	if err != nil {
-		fmt.Fprintln(stdout, err.Error())
+		fmt.Printf(err.Error())
 		os.Exit(1)
 	}
-	fmt.Fprintf(stdout, "Using driver %s\n", driver)
+	fmt.Printf("Using driver %s", driver)
```

Comment:
Printf doesn't add a \n, so you probably need to add it yourself.

Reply:
done